eng/Packages.Data.props

```diff
@@ -38,7 +38,7 @@
     <PackageReference Update="System.Globalization" Version="4.3.0" />
     <PackageReference Update="System.IdentityModel.Tokens.Jwt" Version="5.4.0" />
     <PackageReference Update="System.Linq" Version="4.3.0" />
-    <PackageReference Update="System.Memory.Data" Version="1.0.2" />
+    <PackageReference Update="System.Memory.Data" Version="8.0.0" />
```

Comment:
We cannot arbitrarily upgrade package dependencies - we have to be aware of the versions Functions and PowerShell use to prevent runtime errors from trying to load multiple versions into the same environment.

Reply:
We should consider adding a comment above these package references that links to the specific guidance for vetting interactions with Functions, PowerShell, etc.

Reply:
I like that idea @christothes - would you be willing to put up a PR for it?

Reply:
I created https://github.com/Azure/azure-sdk-for-net/issues/40978 to track this. I think we have some folks that have recently investigated this that might be better suited to pick it up.

Reply:
Thanks @annelo-msft  @christothes . But we need to use the 'MediaType' property of `BinaryData` which is released in latest `System.Memory.Data` version 8.0.0. 
So what's the process to upgrade the dependency library? Thanks

Reply:
Unfortunately, we don't have a streamlined process around this, but @JoshLove-msft should be able to work with you to understand what version Functions is on, and @m-redding may be able to help on the PowerShell side?

Reply:
I can't find anything indicating that PowerShell has a dependency on this package, so I don't think this update would cause versioning issues there.
- https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/System.Management.Automation.csproj
- https://github.com/PowerShell/PowerShell/blob/master/tools/cgmanifest.json
- https://grep.app/search?q=system.memory.data&filter[repo][0]=PowerShell/PowerShell

Reply:
Also I don't have any context around this PR but just wanted to check - this is changing the reference in the Track 1 dependencies list, was that intentional?

Reply:
Thank you @m-redding!!  Super helpful.  I may see if I can do this change in the new System.ClientModel PR given @KrzysztofCwalina's comment: https://github.com/Azure/azure-sdk-for-net/pull/41016#discussion_r1442369969

@JoshLove-msft, do you know how we can understand if this would be a problem for Functions?

Reply:
Functions runs against `net6.0` and, for the in-proc model, will silently downgrade dependencies that the host has.   We should stick with a package in the 6.0 line, if possible.  If we need this, we'll have to run it past Fabio Cavalante to ensure that it won't conflict with the host.

Reply:
We shouldn't be bumping the T1 libraries section regardless.   These bumps should be made only in the T2 sections.